### ISSUE TITLE
Tests Fail when using `hardhat-viem` plugin

### ISSUE THREAD

**Issue by zdenham** [sc-forks/solidity-coverage#874]
### Workaround: (maintainer edit for visibility)

As of **0.8.12**, viem should work with coverage if you set an environment variable before the command as below:
```
SOLIDITY_COVERAGE=true npx hardhat coverage
```

### ----- Original Issue -----

Using hardhat viem:

```typescript
const token = await hre.viem.getContractAt(
    'MyContract',
    contractAddress,
    {
        walletClient,
    }
  );
```

Overriding the wallet client to test calling a contract from a different EOA doesn't seem to work properly (only when using solidity coverage)

```typescript
await token.write.mint(); // does not call with the context of the `walletClient`
```

**Comment by zdenham** [comment]
Here is minimal reproducible example:

https://github.com/zdenham/hardhat-testy-test

**Comment by cgewecke** [comment]
@zdenham Yes, sorry this is known issue with the viem plugin that's being tracked at hardhat in [PR 4961 comment][1]. 


[1]: https://github.com/NomicFoundation/hardhat/pull/4961#issuecomment-1979131000

**Comment by zdenham** [comment]
Got it--appreciate the response @cgewecke. Will keep up to date there

**Comment by joelamouche** [comment]
I have this problem too. Please keep me posted 🙏

**Comment by cgewecke** [comment]
@zdenham, @joelamouche A workaround for this problem has been added in the latest version (0.8.12) via #883

Everything should work as expected if you set a SOLIDITY_COVERAGE env variable to `true` before running the task, e.g"

```
SOLIDITY_COVERAGE=true npx hardhat coverage
```

(This allows the plugin the configure the network correctly before the hardhat-viem plugin hijacks the provider).

Have edited the original issue with this workaround for visibility and am pinning this issue. Please just lmk if this fix does not work for some reason.

**Comment by cgewecke** [comment]
Closing - the current fix is likely permanent.
